services_app/src/androidTest/java/org/opendatakit/utilities/AbstractPermissionsTestCase.java

```diff
@@ -344,29 +345,35 @@ protected ContentValues buildServerRowContent(String tableId, String rowId,
         SyncState.deleted.name() : SyncState.changed.name()));
 
     RowFilterScope.Access localType;
-    if ( rowId.equals(rowIdFullNull) ) {
-      localType = RowFilterScope.Access.FULL;
-      cvValues.put(DataTableColumns.DEFAULT_ACCESS, localType.name());
-      cvValues.putNull(DataTableColumns.ROW_OWNER);
-    } else if ( rowId.equals(rowIdFullCommon) ) {
-      localType = RowFilterScope.Access.FULL;
-      cvValues.put(DataTableColumns.DEFAULT_ACCESS, localType.name());
-      cvValues.put(DataTableColumns.ROW_OWNER, commonUser);
-    } else if ( rowId.equals(rowIdHiddenCommon) ) {
-      localType = RowFilterScope.Access.HIDDEN;
-      cvValues.put(DataTableColumns.DEFAULT_ACCESS, localType.name());
-      cvValues.put(DataTableColumns.ROW_OWNER, commonUser);
-    } else if ( rowId.equals(rowIdReadOnlyCommon) ) {
-      localType = RowFilterScope.Access.READ_ONLY;
-      cvValues.put(DataTableColumns.DEFAULT_ACCESS, localType.name());
-      cvValues.put(DataTableColumns.ROW_OWNER, commonUser);
-    } else if ( rowId.equals(rowIdModifyCommon) ) {
-      localType = RowFilterScope.Access.MODIFY;
-      cvValues.put(DataTableColumns.DEFAULT_ACCESS, localType.name());
-      cvValues.put(DataTableColumns.ROW_OWNER, commonUser);
-    } else {
-      throw new IllegalArgumentException("unexpected rowId value");
-    }
+      switch (rowId) {
```

Comment:
I don't think you should change this to a switch statement. Switch statements are generally should be used on simple data types that you would use the "==" operator on. 

"A switch works with the byte, short, char, and int primitive data types. It also works with enumerated types (discussed in Enum Types), the String class, and a few special classes that wrap certain primitive types: Character, Byte, Short, and Integer (discussed in Numbers and Strings)."
https://docs.oracle.com/javase/tutorial/java/nutsandbolts/switch.html

rowId happens to be a string right now, but we could change that and it would be correct as rowId.equals(rowIdFullNull) as it uses .equals vs "=="

Reply:
Noted. Thank you. I made the changes and pushed.